packages/create-turbo/src/utils/examples.ts

```diff
@@ -0,0 +1,135 @@
+import got from "got";
+import tar from "tar";
+import { Stream } from "stream";
+import { promisify } from "util";
+import { join } from "path";
+import { tmpdir } from "os";
+import { createWriteStream, promises as fs } from "fs";
+
+const pipeline = promisify(Stream.pipeline);
+
+export type RepoInfo = {
+  username: string;
+  name: string;
+  branch: string;
+  filePath: string;
+};
+
+export async function isUrlOk(url: string): Promise<boolean> {
+  try {
+    const res = await got.head(url);
+    return res.statusCode === 200;
+  } catch (err) {
+    return false;
+  }
+}
+
+export async function getRepoInfo(
+  url: URL,
+  examplePath?: string
+): Promise<RepoInfo | undefined> {
+  const [, username, name, t, _branch, ...file] = url.pathname.split("/");
+  const filePath = examplePath
+    ? examplePath.replace(/^\//, "")
+    : file.join("/");
+
+  if (
+    // Support repos whose entire purpose is to be a Turborepo example, e.g.
+    // https://github.com/:username/:my-cool-turborepo-example-repo-name.
+    t === undefined ||
+    // Support GitHub URL that ends with a trailing slash, e.g.
+    // https://github.com/:username/:my-cool-turborepo-example-repo-name/
+    // In this case "t" will be an empty string while the turbo part "_branch" will be undefined
+    (t === "" && _branch === undefined)
+  ) {
+    try {
+      const infoResponse = await got(
+        `https://api.github.com/repos/${username}/${name}`
+      );
+      const info = JSON.parse(infoResponse.body);
+      return { username, name, branch: info["default_branch"], filePath };
+    } catch (err) {
+      return;
```

Comment:
Above, if this fails, we error out that the url is invalid. We should probably change that.

Reply:
👍🏼 updated the error